test/model/internal_link_test.dart

```diff
@@ -0,0 +1,251 @@
+
+import 'package:checks/checks.dart';
+import 'package:test/scaffolding.dart';
+import 'package:zulip/api/model/model.dart';
+import 'package:zulip/model/internal_link.dart';
+import 'package:zulip/model/narrow.dart';
+import 'package:zulip/model/store.dart';
+
+import '../example_data.dart' as eg;
+import 'binding.dart';
+
+Future<PerAccountStore> setupStore(Uri realmUrl, {List<ZulipStream>? streams, User? selfUser}) async {
+  addTearDown(testBinding.reset);
+  selfUser ??= eg.selfUser;
+  streams ??= [];
+  final account = Account(
+    id: 1001,
+    realmUrl: realmUrl,
+    email: selfUser.email,
+    userId: selfUser.userId,
+    apiKey: 'abcd',
+    zulipFeatureLevel: eg.recentZulipFeatureLevel,
+    zulipVersion: eg.recentZulipVersion,
+    zulipMergeBase: eg.recentZulipVersion,
+  );
+  final initialSnapshot = eg.initialSnapshot(streams: streams);
+  await testBinding.globalStore.add(account, initialSnapshot);
+  return await testBinding.globalStore.perAccount(account.id);
```

Comment:
We shouldn't need the global binding here; we're just passing stores around explicitly. See other test files in `test/model/` for examples.